From 101b02d2849c7f945ffc5971f34246a2e0aac366 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Fri, 15 Sep 2017 14:00:38 +0700
Subject: [PATCH 323/433] drm: fb_helper: Fix return value of
 drm_fb_helper_is_bound

drm_fb_helper_is_bound should not judge false when the frame
buffer which is set to each CRTCs differs in R-Car driver.
This patch corrects so that true may always be returned in
R-Car driver.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
(cherry picked from commit d634ef261b74b4a2bf97dcdd78885787cde5ad89)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 drivers/gpu/drm/drm_fb_helper.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/drm_fb_helper.c b/drivers/gpu/drm/drm_fb_helper.c
index edec88b..dc6bfcd 100644
--- a/drivers/gpu/drm/drm_fb_helper.c
+++ b/drivers/gpu/drm/drm_fb_helper.c
@@ -483,6 +483,7 @@ EXPORT_SYMBOL(drm_fb_helper_restore_fbdev_mode_unlocked);
 
 static bool drm_fb_helper_is_bound(struct drm_fb_helper *fb_helper)
 {
+#if !defined(CONFIG_DRM_RCAR_DU) && !defined(CONFIG_DRM_RCAR_DU_MODULE)
 	struct drm_device *dev = fb_helper->dev;
 	struct drm_crtc *crtc;
 	int bound = 0, crtcs_bound = 0;
@@ -501,7 +502,7 @@ static bool drm_fb_helper_is_bound(struct drm_fb_helper *fb_helper)
 
 	if (bound < crtcs_bound)
 		return false;
-
+#endif
 	return true;
 }
 
-- 
2.7.4

